thunder/torch/__init__.py

```diff
@@ -3005,6 +3005,52 @@ def topk(
     return clang.topk(a, k, dim, largest, sorted, out=out)
 
 
+@torchsymbol(torch.atleast_1d, is_method=True)
+def atleast_1d(
+    arg: Union[TensorLike, Sequence[TensorLike]], *args: TensorLike
+) -> Union[TensorLike, tuple[TensorLike, ...]]:
+    if not args and isinstance(arg, Sequence):
+        args_ = arg
+    else:
+        assert not isinstance(arg, Sequence)
+        args_ = (arg,) + args
+    res = tuple(a if a.ndim >= 1 else unsqueeze(a, 0) for a in args_)
+    return res if len(res) > 1 else res[0]
+
+
+def _unsqueeze_atleast(at_least_fn: Callable, dim: int, arg: TensorLike) -> TensorLike:
+    arg_ = at_least_fn(arg)
+    return unsqueeze(arg_, dim)
+
+
+@torchsymbol(torch.atleast_2d, is_method=True)
+def atleast_2d(
+    arg: Union[TensorLike, Sequence[TensorLike]], *args: TensorLike
+) -> Union[TensorLike, tuple[TensorLike, ...]]:
+    if not args and isinstance(arg, Sequence):
+        args_ = arg
+    else:
+        assert not isinstance(arg, Sequence)
+        args_ = (arg,) + args
+    unsqueeze_atleast_1d = partial(_unsqueeze_atleast, atleast_1d, 0)
+    res = tuple(a if a.ndim >= 2 else unsqueeze_atleast_1d(a) for a in args_)
```

Comment:
Instead of using `unsqueeze_atleast` as a helper, what would this look like if you defined a function within `atleast_2d` that handled the two unsqueezing cases?

```
if a.ndims == 0:
  return a.unsqueeze(0, 1)
if a.ndims == 1:
  return a.unsqueeze(0)
return a
```

?

I think this would be more readable and ensure the generated trace is as simple as possible

Reply:
Make sense! Definitely a better way. Thank you! I've updated the code accordingly.